src/test/java/md/maib/service/impl/CustomerServiceImplTest.java

```diff
@@ -0,0 +1,116 @@
+package md.maib.service.impl;
+
+import md.maib.entity.Customer;
+import md.maib.repository.CustomerRepository;
+import md.maib.service.CustomerService;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CustomerServiceImplTest {
+
+    static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:15-alpine");
+
+    @Autowired
+    private CustomerService customerService;
+
+    @Autowired
+    private CustomerRepository customerRepository;
+
+    private Customer savedCustomer = null;
+
+    @BeforeAll
+    static void beforeAll() {
+        postgres.start();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        postgres.stop();
+    }
+
+    @DynamicPropertySource
+    static void configureProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", postgres::getJdbcUrl);
+        registry.add("spring.datasource.username", postgres::getUsername);
+        registry.add("spring.datasource.password", postgres::getPassword);
+    }
+
+    @BeforeEach
+    public void initCustomer() {
+        customerRepository.deleteAll();
+        savedCustomer = new Customer.Builder()
+                .firstName("John")
+                .lastName("Doe")
+                .pan("1234567891011121")
+                .cvv("123")
+                .build();
+
+        customerRepository.save(savedCustomer);
+    }
+
+    @Test
+    void shouldGetAllCustomers() {
+        List<Customer> customers = customerService.getAllCustomers();
+        assertFalse(customers.isEmpty(), "The customer list should not be empty");
+        assertEquals(1, customers.size(), "The customer list should contain exactly one customer");
+    }
+
+    @Test
+    void shouldGetCustomerById() {
+        var customerId = customerService.getCustomerById(savedCustomer.getId())
+                .orElseThrow(() -> new RuntimeException("Customer not found"))
```

Comment:
Test should verify 1 case. One test -  one assert. You also have here exception scenario. Please exclude.